src/CachePeer.cc

```diff
@@ -114,6 +117,208 @@ CachePeer::connectTimeout() const
+
+    if (options.mcast_responder)
+        os << " multicast-responder";
+
+#if PEER_MULTICAST_SIBLINGS
+    if (options.mcast_siblings)
+        os << " multicast-siblings";
+#endif
+
+    if (weight != 1)
+        os << " weight=" << weight;
+
+    if (options.closest_only)
+        os << " closest-only";
+
+#if USE_HTCP
+    if (options.htcp)
+    {
+        std::vector<const char *> opts {};
+        if (options.htcp_oldsquid)
+            opts.push_back("oldsquid");
+        if (options.htcp_no_clr)
+            opts.push_back("no-clr");
+        if (options.htcp_no_purge_clr)
+            opts.push_back("no-purge-clr");
+        if (options.htcp_only_clr)
+            opts.push_back("only-clr");
+        os << AsList(opts).prefixedBy(" htcp=").delimitedBy(",");
+    }
+#endif
+
+    if (options.no_netdb_exchange)
+        os << " no-netdb-exchange";
+
+#if USE_DELAY_POOLS
+    if (options.no_delay)
+        os << " no-delay";
+#endif
+
+    if (login)
+        os << " login=" << login;
+
+    if (mcast.ttl > 0)
+        os << " ttl=" << mcast.ttl;
+
+    if (connect_timeout_raw > 0)
+        os << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        os << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+    if (digest_url)
+        os << " digest-url=" << digest_url;
+#endif
+
+    if (options.allow_miss)
+        os << " allow-miss";
+
+    if (options.no_tproxy)
+        os << " no-tproxy";
+
+    if (max_conn > 0)
+        os << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        os << " standby=" << standby.limit;
+
+    if (options.originserver)
+        os << " originserver";
+
+    if (domain)
+        os << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        os << " connection-auth=off";
+    else if (connection_auth == 1)
+        os << " connection-auth=on";
+    else if (connection_auth == 2)
+        os << " connection-auth=auto";
+
+    secure.dumpCfg(os, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& os)
+{
+    os <<
+       "- " << name << ":\n" <<
+       "  type: " << typeString() << '\n' <<
+       "  host: " << host << '/' << http_port << '/' << icp.port << '\n';
+    os << "  flags:";
+    dumpOptions(os);
+    os << '\n';
+
+    std::vector<Ip::Address> addr;
+    std::copy(addresses, addresses+n_addresses, std::back_inserter(addr));
+    os << "  addresses: [ " << AsList(addr).prefixedBy("\"").delimitedBy("\", \"").suffixedBy("\"") << " ]\n";
+
+    os << "  status: " << (neighborUp(this) ? "Up" : "Down") << '\n' <<
+       "  fetches: " << stats.fetches << '\n' <<
+       "  open connections: " << stats.conn_open << '\n' <<
+       "  average RTT: " << stats.rtt << " msec\n";
+
+    if (stats.last_query > 0) {
+        os << "  last query:" <<
+           (squid_curtime - stats.last_query) << " seconds ago\n";
+    }
+
+    if (stats.last_reply > 0) {
+        os << " last reply: " <<
```

Comment:
Missing one space in indentation? Should we add a yamlIndent(n = 2) manipulator to make this indentation more "explicit" and harder to screw up?

```suggestion
        os << yamlIndent << "last reply: " <<
```

Those lines that need to double the indentation can use either

```C++
        os << yamlIndent(4) << "htcp hits: " << ...
```

or (to reduce work for those implementing yamlIndent at the expense of callers):

```C++
        os << yamlIndent << yamlIndent << "htcp hits: " << ...
```

or, with more work for those implementing yamlIndent, even

```C++
        os << yamlIndent*2 << "htcp hits: " << ...
```

Reply:
I thought about that, but didn't want to go there yet
I would call it indent() or spaces(), no need to be too specific here

Reply:
> I thought about that, but didn't want to go there yet

Indentation mistakes in this PR are enough reasons to add/use yamlIndent(), indent(), or spaces() IMO.

>  I would call it indent() or spaces(), no need to be too specific here

* yamlIndent() looks ugly but makes it easy to select the default number of spaces specifically for YAML output.
* indent() looks much better than yamlIndent(), but I doubt we will have two-space indentation in many contexts other than YAML. That is why I though of yamlIndent()... If we switch to four-space indentation for YAML, then I am OK with indent() or indent(n = 4) because that default works great in most text formatting contexts.
* spaces(n) look great and is more versatile, but having a default parameter for a function called spaces() is a bit odd IMHO so we would have to always say spaces(2) or spaces(4) explicitly for default indentation cases. This is acceptable IMO, but not ideal.

Reply:
I'd go for spaces(n) as it keeps things simple and is highly visible.
If we wanted to have something stateful, it could be an indent object, with up() and down() methods, but is it really worth it?

Reply:
> I'd go for spaces(n) as it keeps things simple and is highly visible.

I assume you mean spaces() function _without_ a default value for its `n` parameter. I am OK with that option.

Please make it a constexpr function that only takes one parameter and returns a string literal, a `const char *` pointer, or equivalent. If you need help with that, please ask! Otherwise, we would have to use a stream manipulator pattern which is a lot of work producing worse result (please do not add an SBuf-returning or similar heavy/noisy/memory-allocating function!).